routers/web/auth/oauth.go

```diff
@@ -1041,7 +1041,7 @@ func handleOAuth2SignIn(ctx *context.Context, source *auth.Source, u *user_model
 		}
 
 		// update external user information
-		if err := externalaccount.UpdateExternalUser(u, gothUser); err != nil {
+		if err := externalaccount.UpdateExternalUser(u, gothUser, false); err != nil {
```

Comment:
why upsert=false?

Reply:
Unlike `handleUserCreated`, in `handleOAuth2SignIn`, the external user is expected to exist.

Reply:
Hmm, I tried to read the code, but the flow is quite complex.

I am not sure that handleOAuth2SignIn is guaranteed to be called after external user is prepared correctly.

Since the code was that before, I think there shouldn't be more problem. So LGTM.

Reply:
Actually, I'm not even sure the stored external user data is being used. The code couldn't store any data before, but it worked fine, just caused user concern because of error logs.